pkg/replicate/scheme.go

```diff
@@ -117,29 +118,13 @@ type replicationScheme struct {
 }
 
 type replicationMetrics struct {
-	originIterations  prometheus.Counter
-	originMetaLoads   prometheus.Counter
-	originPartialMeta prometheus.Counter
-
 	blocksAlreadyReplicated prometheus.Counter
 	blocksReplicated        prometheus.Counter
 	objectsReplicated       prometheus.Counter
 }
 
 func newReplicationMetrics(reg prometheus.Registerer) *replicationMetrics {
 	m := &replicationMetrics{
-		originIterations: promauto.With(reg).NewCounter(prometheus.CounterOpts{
-			Name: "thanos_replicate_origin_iterations_total",
-			Help: "Total number of objects iterated over in the origin bucket.",
-		}),
-		originMetaLoads: promauto.With(reg).NewCounter(prometheus.CounterOpts{
```

Comment:
Do we really want to remove this and the following metric? May still be useful. The previous one definitely needs to go, though.

Reply:
Thanks for the quick review! This one `thanos_replicate_origin_meta_loads_total` is the same as the fetcher metric `thanos_blocks_meta_synced{state="loaded"}`

Reply:
Yep! You are correct. Let's document that in the `CHANGELOG.md` file and it should be good to go :+1:

Reply:
By removing `thanos_replicate_origin_meta_loads_total`   and replacing it with `thanos_blocks_meta_synced` there is no way to distinct metrics from the 2 fetchers 🤔

Reply:
So we either need to put back the origin metrics, or expose a label in the fetchers to distinguish them. E.g bucket name.

Reply:
Sorry, I didn't get the point. What do you mean by `distinct metrics from the 2 fetchers`? If the fetcher metrics are exposed by 2 components, then they must have different job labels? Maybe I misunderstood your question. WDYT?

Reply:
> If the fetcher metrics are exposed by 2 components
this would be correct, but not in this case. In this case 2 fetchers are being used in the same component. Fetchers for the 2 buckets.

Reply:
But maybe it is still save the expose the bucket name as a static label in the fetcher metrics?

Reply:
I don't know the answer. But this is not part of this pr and fetcher metrics are used in store and compactor as well, so if you think it is necessary to have another label we could have another issue to track it.

Reply:
Yup, let's propose this in other Pr if needed